modules/diskq/tests/test_diskq.c

```diff
@@ -315,6 +317,73 @@ testcase_with_threads(void)
   fprintf(stderr, "Feed speed: %.2lf\n", (double) TEST_RUNS * MESSAGES_SUM * 1000000 / sum_time);
 }
 
+#define TEST_DISK_BUF_SIZE  1000123
+#define TEST_MEM_SIZE  1000123
+static void
+testcase_diskbuffer_restart_corrupted(void)
+{
+  typedef struct restart_test_parameters
+  {
+    gchar *filename;
+    LogQueue *(*lqdisk_new_func)(DiskQueueOptions *options, const gchar *persist_name);
+    gboolean reliable;
+  } restart_test_parameters;
+
+  restart_test_parameters parameters[] =
+  {
+    {"test-diskq-restart.qf", log_queue_disk_non_reliable_new, FALSE},
+    {"test-diskq-restart.rqf", log_queue_disk_reliable_new, TRUE},
+  };
+
+  gint number_of_variants = sizeof(parameters)/sizeof(restart_test_parameters);
+  for (gint i = 0; i < number_of_variants; i++)
+    {
+      DiskQueueOptions options;
+      _construct_options(&options, TEST_DISK_BUF_SIZE, TEST_MEM_SIZE, parameters[i].reliable);
+      LogQueue *q = parameters[i].lqdisk_new_func(&options, NULL);
+      log_queue_set_use_backlog(q, FALSE);
+
+      gchar *filename = parameters[i].filename;
+      gchar filename_corrupted_dq[100];
+      g_snprintf(filename_corrupted_dq, 100, "%s.corrupted", filename);
+      unlink(filename);
+      unlink(filename_corrupted_dq);
+
+      log_queue_disk_load_queue(q, filename);
+      fed_messages = 0;
+      feed_some_messages(q, 100, &parse_options);
+      assert_gint(fed_messages, 100, "Failed to push all messages to the disk-queue!\n");
+
+      LogQueueDisk *disk_queue = (LogQueueDisk *)q;
+      disk_queue->restart_corrupted(disk_queue);
+
+      struct stat file_stat;
+      stat(filename, &file_stat);
+      assert_gint(S_ISREG(file_stat.st_mode), TRUE,
+                  "New disk-queue file does not exists!!");
+      stat(filename_corrupted_dq, &file_stat);
+      assert_gint(S_ISREG(file_stat.st_mode), TRUE,
+                  "Corrupted disk-queue file does not exists!!");
+      assert_string(qdisk_get_filename(disk_queue->qdisk), filename,
+                    "New disk-queue file's name should be the same\n");
+      assert_gint(qdisk_get_size(disk_queue->qdisk), TEST_DISK_BUF_SIZE,
+                  "Disk-queue option does not match the original configured value!\n");
+      assert_gint(qdisk_get_length(disk_queue->qdisk), 0,
+                  "New disk-queue file should be empty!\n");
+      assert_gint(qdisk_get_writer_head(disk_queue->qdisk), QDISK_RESERVED_SPACE,
+                  "Invalid write pointer!\n");
+      assert_gint(qdisk_get_reader_head(disk_queue->qdisk), QDISK_RESERVED_SPACE,
+                  "Invalid read pointer!\n");
+
+      log_queue_unref(q);
+      unlink(filename);
+      unlink(filename_corrupted_dq);
+      disk_queue_options_destroy(&options);
+    }
+}
+#undef TEST_DISK_BUF_SIZE
+#undef TEST_MEM_SIZE
```

Comment:
I wouldn't unset these macros.

Reply:
Why not simple use a `const` variable ?

Reply:
`TEST_MEM_SIZE` is not even used for assertion as checking `disk_buf_size` is good enough.
I prefer macros as constants, but for here the const var is good too.
I have changed it to const var. and moved it inside the function.